Vungle/VungleAdapter/ALVungleMediationAdapter.m

```diff
@@ -893,80 +921,73 @@ - (instancetype)initWithParentAdapter:(ALVungleMediationAdapter *)parentAdapter
+    if ( !CGSizeEqualToSize(CGSizeZero, adSize) )
     {
-        [self.delegate didLoadAdForAdView: self.parentAdapter.adViewContainer];
+        extraInfo[@"ad_width"] = @(adSize.width);
+        extraInfo[@"ad_height"] = @(adSize.height);
     }
     
-    if ( [banner canPlayAd] )
-    {
-        [banner presentOn: self.parentAdapter.adViewContainer];
-    }
-    else
-    {
-        [self.parentAdapter log: @"Failed to load ad view ad: ad not ready"];
-        [self.delegate didFailToLoadAdViewAdWithError: MAAdapterError.adNotReady];
-    }
+    [self.delegate performSelector: @selector(didLoadAdForAdView:withExtraInfo:)
+                        withObject: bannerView
+                        withObject: extraInfo];
 }
 
-- (void)bannerAdDidFailToLoad:(VungleBanner *)banner withError:(NSError *)error
+- (void)bannerAdDidFail:(VungleBannerView *)bannerView withError:(NSError *)error
 {
     MAAdapterError *adapterError = [ALVungleMediationAdapter toMaxError: error isAdPresentError: NO];
     [self.parentAdapter log: @"AdView failed to load with error: %@", adapterError];
     [self.delegate didFailToLoadAdViewAdWithError: adapterError];
 }
 
-- (void)bannerAdWillPresent:(VungleBanner *)banner
+- (void)bannerAdWillPresent:(VungleBannerView *)bannerView
 {
-    [self.parentAdapter log: @"AdView ad will present %@", banner.placementId];
+    [self.parentAdapter log: @"AdView ad will present %@", bannerView.placementId];
 }
 
-- (void)bannerAdDidPresent:(VungleBanner *)banner
+- (void)bannerAdDidPresent:(VungleBannerView *)bannerView
 {
-    [self.parentAdapter log: @"AdView ad shown %@", banner.placementId];
+    [self.parentAdapter log: @"AdView ad shown %@", bannerView.placementId];
 }
 
-- (void)bannerAdDidTrackImpression:(VungleBanner *)banner
+- (void)bannerAdDidTrackImpression:(VungleBannerView *)bannerView
 {
-    [self.parentAdapter log: @"AdView ad impression tracked %@", banner.placementId];
+    [self.parentAdapter log: @"AdView ad impression tracked %@", bannerView.placementId];
     [self.delegate didDisplayAdViewAd];
 }
 
-- (void)bannerAdDidClick:(VungleBanner *)banner
+- (void)bannerAdDidClick:(VungleBannerView *)bannerView
 {
-    [self.parentAdapter log: @"AdView ad clicked %@", banner.placementId];
+    [self.parentAdapter log: @"AdView ad clicked %@", bannerView.placementId];
     [self.delegate didClickAdViewAd];
 }
 
-- (void)bannerAdWillLeaveApplication:(VungleBanner *)banner
+- (void)bannerAdWillLeaveApplication:(VungleBannerView *)bannerView
 {
-    [self.parentAdapter log: @"AdView ad will leave application %@", banner.placementId];
-}
-
-- (void)bannerAdDidFailToPresent:(VungleBanner *)banner withError:(NSError *)error
-{
-    MAAdapterError *adapterError = [ALVungleMediationAdapter toMaxError: error isAdPresentError: YES];
-    [self.parentAdapter log: @"AdView ad failed to present with error: %@", adapterError];
-    [self.delegate didFailToDisplayAdViewAdWithError: adapterError];
```

Comment:
u need to move this logic into bannerAdDidFail: callback.

Reply:
I have below in `bannerAdDidFail`. 

```
    MAAdapterError *adapterError = [ALVungleMediationAdapter toMaxError: error isAdPresentError: NO];
    [self.parentAdapter log: @"AdView failed to load with error: %@", adapterError];
    [self.delegate didFailToLoadAdViewAdWithError: adapterError];
 ```
 
 Do you know if we need `isAdPresentError: NO` or `isAdPresentError: YES` ?

Reply:
its based on present error or load error.

Reply:
Updated.